src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -288,6 +298,46 @@ public static int getFuelValue(ItemStack itemStack)
         }
         return fuelValue;
     }
+    
+    /**
+     * Links a set of parts to an {@link IPartFactory} that can produce them.
+     */
+    public static void registerPartFactory(IPartFactory factory, ResourceLocation... names)
+    {
+        MultipartRegistry.registerPartFactory(factory, names);
+    }
+    
+    /**
+     * Links a set of parts to an {@link IAdvancedPartFactory} that can produce them.
+     */
+    public static void registerPartFactory(IAdvancedPartFactory factory, ResourceLocation... names)
```

Comment:
Why are you adding names? What are these functions doing? also why do you need a IPartFactory and an IAdvancedPartFactory?
We're trying to move away from register*** functions and just use the register(IForgeRegistryEntry)